server/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -109,10 +114,36 @@ public QueryRewriteContext(final XContentParserConfiguration parserConfiguration
             null,
             null,
             null,
+            null,
             null
         );
     }
 
+    public QueryRewriteContext(
```

Comment:
I guess we can leave this out for another PR, but could we potentially extract this constructor to a new class that doesn't need all the other cruft?

Reply:
Agreed this could use a good refactor, but IMO it's best left for another PR